File: backend/routers/system.py
import os
import threading
import time

from fastapi import APIRouter

router = APIRouter(tags=["system"])


# Shutdown route for the PC app to close the server
@router.post("/shutdown")
def shutdown():
    def stop():
        time.sleep(0.5)
        os._exit(0)

    threading.Thread(target=stop).start()
    return {"message": "Server shutting down"}

File: backend/src/routers/system.rs
use axum::Json;
use serde_json::{json, Value};
use std::time::Duration;
use tokio::time::sleep;

pub async fn shutdown() -> Json<Value> {
    tokio::spawn(async {
        sleep(Duration::from_millis(500)).await;
        std::process::exit(0);
    });
    Json(json!({ "message": "Server shutting down" }))
}
